pkg/util/json/json_test.go

```diff
@@ -881,6 +881,95 @@ func TestEncodeDecodeJSONInvertedIndex(t *testing.T) {
 	}
 }
 
+const expectError = "<<error>>"
+
+func TestConcat(t *testing.T) {
+	cases := map[string][]struct {
+		concatWith string
+		expected   string
+	}{
+		`null`: {
+			{`null`, `[null, null]`},
+			{`[]`, `[null]`},
+			{`[1]`, `[null, 1]`},
+			{`{}`, expectError},
+		},
+		`true`: {
+			{`null`, `[true, null]`},
+			{`[]`, `[true]`},
+			{`[1]`, `[true, 1]`},
+			{`{}`, expectError},
+		},
+		`false`: {
+			{`null`, `[false, null]`},
+			{`[]`, `[false]`},
+			{`[1]`, `[false, 1]`},
+			{`{}`, expectError},
+		},
+		`123`: {
+			{`null`, `[123, null]`},
+			{`[]`, `[123]`},
+			{`[1]`, `[123, 1]`},
+			{`{}`, expectError},
```

Comment:
What about concatting a scalar?
```
jordan=# select '1'::jsonb || '1';
?column?

Reply:
`null` is a scalar! But I've added some other scalars as well.